tests/snakemake/__init__.py

```diff
@@ -102,41 +104,42 @@ def compare_matrices(
     run.cache_comparisons()  # could be all NaN if constructed in steps
     assert run.identities is not None
     assert matrices_path.is_dir()
+    method = run.configuration.method
 
     checked = False
-    if (matrices_path / "matrix_identity.tsv").is_file():
+    if (matrices_path / f"{method}_identity.tsv").is_file():
         compare_matrix(
             run.identities,
-            matrices_path / "matrix_identity.tsv",
+            matrices_path / f"{method}_identity.tsv",
             absolute_tolerance=absolute_tolerance,
         )
         checked = True
-    if (matrices_path / "matrix_aln_lengths.tsv").is_file():
-        compare_matrix(run.aln_length, matrices_path / "matrix_aln_lengths.tsv")
+    if (matrices_path / f"{method}_aln_lengths.tsv").is_file():
+        compare_matrix(run.aln_length, matrices_path / f"{method}_aln_lengths.tsv")
         checked = True
-    if (matrices_path / "matrix_coverage.tsv").is_file():
+    if (matrices_path / f"{method}_coverage.tsv").is_file():
         compare_matrix(
             run.cov_query,
-            matrices_path / "matrix_coverage.tsv",
+            matrices_path / f"{method}_coverage.tsv",
             absolute_tolerance=absolute_tolerance,
         )
         checked = True
-    if (matrices_path / "matrix_hadamard.tsv").is_file():
+    if (matrices_path / f"{method}_hadamard.tsv").is_file():
         compare_matrix(
             run.hadamard,
-            matrices_path / "matrix_hadamard.tsv",
+            matrices_path / f"{method}_hadamard.tsv",
             absolute_tolerance=absolute_tolerance,
         )
         checked = True
-    if (matrices_path / "matrix_sim_errors.tsv").is_file():
-        if "dnadiff" in str(matrices_path):
+    if (matrices_path / f"{method}_sim_errors.tsv").is_file():
+        if method == "dnadiff" and "/viral_example/matrices" in str(matrices_path):
```

Comment:
This special case (which I made a little stricter) stems from #148 and is a workarond for this difference:

Expected output directly from parsing dnadiff:

```
❯ xsv table tests/fixtures/viral_example/matrices/dnadiff_sim_errors.tsv
                                 5584c7029328dc48d33f95f0a78f7e57 689d3fd6881db36b5e08329cf23cecdd 78975d5144a1cd12e98898d573cf6536
5584c7029328dc48d33f95f0a78f7e57 0                                27                               214
689d3fd6881db36b5e08329cf23cecdd 27                               0                                145
78975d5144a1cd12e98898d573cf6536 214                              145                              0
```

Actual output via parsing intermediate files:

```
                                 5584c7029328dc48d33f95f0a78f7e57 689d3fd6881db36b5e08329cf23cecdd 78975d5144a1cd12e98898d573cf6536
5584c7029328dc48d33f95f0a78f7e57 0                                27                               214
689d3fd6881db36b5e08329cf23cecdd 27                               0                                146
78975d5144a1cd12e98898d573cf6536 214                              146                              0
```

Off by one, 145 vs 156.